lib/secretsManager.js

```diff
@@ -1,77 +1,161 @@
-let DUMMY_UUID_GEN = 0
-let DUMMY_STORE = []
-
-/**
- * Find all secrets, applying a selector if provided.
- * When provided, all where clause must be fulfilled (logical AND).
- * By default, returns all secrets
- * @async
- * @param {Object} where - hash of where clauses returned secrets must fulfill with
- * @returns {Array<Object>} list (may be empty) of matching secrets
- */
-export async function find(where = {}) {
-  let secrets = []
-
-  try {
-    const props = Object.keys(where)
-    secrets = DUMMY_STORE.filter(secret =>
-      props.every(prop => secret[prop] === where[prop])
+import { v4 as uuid } from 'uuid'
+import aesjs from 'aes-js'
+import * as SecureStore from 'expo-secure-store'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+
+const encryptionPassphraseStorageKey = 'encryptionPassphrase'
+const secretStorageKey = 'secrets'
+
+export default class SecretsManager {
+  constructor() {
+    this.encryptionPassphrase = null
+  }
+
+  async initialize() {
+    const [
+      getEncryptionPassphrase,
+      setEncryptionPassphrase,
+    ] = (await SecureStore.isAvailableAsync())
+      ? [SecureStore.getItemAsync, SecureStore.setItemAsync]
+      : [AsyncStorage.getItem, AsyncStorage.setItem]
+
+    const encryptionPassphrase = await getEncryptionPassphrase(
+      encryptionPassphraseStorageKey
+    )
+
+    if (encryptionPassphrase) {
+      this.encryptionPassphrase = JSON.parse(encryptionPassphrase)
+      return
+    }
+
+    this.encryptionPassphrase = this.generateEncryptionPassphrase()
+
+    await this.reset()
```

Comment:
do we really need this if we're going to overwrite whatever was there two lines below?

Reply:
Yeah, this function really started as a solution for me to cleanup the mess while getting the encryption right. I guess we can do good enough by just removing or overwriting the secrets collection here. Thoughts?

Reply:
I agree